src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisConfigDrawer.tsx

```diff
@@ -0,0 +1,191 @@
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import { RecordingAttributes } from '@app/Shared/Services/Api.service';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { automatedAnalysisConfigToRecordingAttributes } from '@app/Shared/Services/Settings.service';
+import { useSubscriptions } from '@app/utils/useSubscriptions';
+import {
+  Drawer,
+  DrawerActions,
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerContentBody,
+  DrawerHead,
+  DrawerPanelBody,
+  DrawerPanelContent,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  DropdownToggleAction,
+  Level,
+  LevelItem,
+} from '@patternfly/react-core';
+import { CogIcon, PlusCircleIcon } from '@patternfly/react-icons';
+import * as React from 'react';
+import { first } from 'rxjs';
+import { AutomatedAnalysisConfigForm } from './AutomatedAnalysisConfigForm';
+
+interface AutomatedAnalysisConfigDrawerProps {
+  drawerContent: React.ReactNode;
+  onCreate: () => void;
+}
+
+export const AutomatedAnalysisConfigDrawer: React.FunctionComponent<AutomatedAnalysisConfigDrawerProps> = (props) => {
+  const context = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+
+  const [isExpanded, setIsExpanded] = React.useState(false);
+  const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+  const drawerRef = React.useRef<HTMLDivElement>(null);
+
+  const onToggle = React.useCallback(
+    (selected) => {
+      setIsDropdownOpen(selected);
+    },
+    [setIsDropdownOpen]
+  );
+
+  const handleCreateRecording = React.useCallback(
+    (recordingAttributes: RecordingAttributes) => {
+      addSubscription(
+        context.api
+          .createRecording(recordingAttributes)
+          .pipe(first())
+          .subscribe({
+            next: (resp) => {
+              if (resp && resp.ok) {
+                props.onCreate();
+              }
+            },
+            error: (err) => {},
+          })
+      );
+    },
+    [addSubscription, context.api, props.onCreate]
+  );
+
+  const onDefaultRecordingStart = React.useCallback(() => {
+    const config = context.settings.automatedAnalysisRecordingConfig();
+    const attributes = automatedAnalysisConfigToRecordingAttributes(config);
+    handleCreateRecording(attributes);
+  }, [context.settings, context.settings.automatedAnalysisRecordingConfig, handleCreateRecording]);
+
+  const onExpand = React.useCallback(() => {
```

Comment:
should put `drawerRef` as deps. Though I don't think it will change. Just for good practice :D